### PR TITLE
Final collection and final set

### PR DESCRIPTION
**Pull request by ZaruhiAvagyan**

### CODE REVIEW

File: src/com/collection/DoublyLinkedList.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
+package com.collection;
+
+public class DoublyLinkedList {
+
+    Node head;
```
Comment (naira-tamrazyan) [inline src/com/collection/DoublyLinkedList.java]: why head is not private?

File: src/com/collection/DoublyLinkedList.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        public void setNext(Node next) { this.next = next; }
+    }
+
+    //inserts at the front
+    public void push(ComparableStudent student){
+
+        Node first = new Node(student);
+        first.setNext(getHead());
+        first.setPrevious(null);
+        if(getHead() != null)
```
Comment (naira-tamrazyan) [inline src/com/collection/DoublyLinkedList.java]: According to Java coding conventions it is good to put curly braces {} event if there is only one statement after if:
   if(getHead() != null) {
      getHead().setPrevious(first);
   }

File: src/com/collection/DoublyLinkedList.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        public Node getPrevious() { return previous; }
+        public void setPrevious(Node previous) { this.previous = previous; }
+        public Node getNext() { return next; }
+        public void setNext(Node next) { this.next = next; }
+    }
+
+    //inserts at the front
+    public void push(ComparableStudent student){
+
+        Node first = new Node(student);
```
Comment (naira-tamrazyan) [inline src/com/collection/DoublyLinkedList.java]: what if student is null?

File: src/com/collection/DynamicArrayForStudents.java [hunk 4]
```diff
@@ -33,14 +33,14 @@ public void setCapacity(int capacity) {
     public DynamicArrayForStudents()
     {
         setCapacity(10);      //let's assume default capacity = 10
```
Comment (naira-tamrazyan) [inline src/com/collection/DynamicArrayForStudents.java]: You may create a constant for 10 and use it in code

File: src/com/collection/Test.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        studentsDynamic.add(new ComparableStudent("Ashot", "Sargsyan", 25));
+        studentsDynamic.add(new ComparableStudent("Susanna", "Gasparyan", 19));
+        studentsDynamic.add(new ComparableStudent("Vahagn", "Harutyunyan", 21));
+
+        System.out.printf("Array size is %d\n", studentsDynamic.getSize());
+        System.out.printf("Array capacity is %d\n", studentsDynamic.getCapacity());
+        DynamicArrayForStudents.print(studentsDynamic);
+
+        try {
+            studentsDynamic.remove(new ComparableStudent("Vahagn", "Harutyunyan", 21));
```
Comment (naira-tamrazyan) [inline src/com/collection/Test.java]: It is good to throw exception in add and  remove methods  when the element is null (for example: IllegalArgumentException) and not catch a NullPointerException here.

File: src/com/collection/Test.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+        System.out.printf("Array size is %d\n", studentsDynamic.getSize());
+        System.out.printf("Array capacity is %d\n", studentsDynamic.getCapacity());
+        DynamicArrayForStudents.print(studentsDynamic);
+        System.out.println();
+
+        //Get the firstName of Student at index=1
+        try {
+            ComparableStudent student = studentsDynamic.get(1);
+        }catch(ArrayIndexOutOfBoundsException e){
```
Comment (naira-tamrazyan) [inline src/com/collection/Test.java]: Same here

**Review by naira-tamrazyan — CHANGES_REQUESTED** [review]

File: src/com/collection/DoublyLinkedList.java (hunk 1 above)
Comment (ZaruhiAvagyan) [reply src/com/collection/DoublyLinkedList.java]: it should be private

File: src/com/collection/DoublyLinkedList.java (hunk 2 above)
Comment (ZaruhiAvagyan) [reply src/com/collection/DoublyLinkedList.java]: okay

File: src/com/collection/DoublyLinkedList.java (hunk 3 above)
Comment (ZaruhiAvagyan) [reply src/com/collection/DoublyLinkedList.java]: throw new IllegalArgumentException("Argument cannot be null");   ??

File: src/com/collection/DynamicArrayForStudents.java (hunk 4 above)
Comment (ZaruhiAvagyan) [reply src/com/collection/DynamicArrayForStudents.java]: okay

File: src/com/collection/Test.java (hunk 5 above)
Comment (ZaruhiAvagyan) [reply src/com/collection/Test.java]: okay

**Review by naira-tamrazyan — APPROVED** [review]

**Pull request merged**